In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv("../data/Dataset.csv")

In [ ]:
df.columns

Index(['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address',
       'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines',
       'Average Cost for two', 'Currency', 'Has Table booking',
       'Has Online delivery', 'Is delivering now', 'Switch to order menu',
       'Price range', 'Aggregate rating', 'Rating color', 'Rating text',
       'Votes'],
      dtype='object')

In [ ]:
df.shape

(9551, 21)

In [ ]:
df.isnull().sum()

Restaurant ID           0
Restaurant Name         0
Country Code            0
City                    0
Address                 0
Locality                0
Locality Verbose        0
Longitude               0
Latitude                0
Cuisines                9
Average Cost for two    0
Currency                0
Has Table booking       0
Has Online delivery     0
Is delivering now       0
Switch to order menu    0
Price range             0
Aggregate rating        0
Rating color            0
Rating text             0
Votes                   0
dtype: int64

In [ ]:
df = df.dropna(subset=["Cuisines"])

# Level 2 Task 2 - Table Booking and Online Delivery

In [ ]:
df["Has Table booking"].value_counts()

Has Table booking
No     8384
Yes    1158
Name: count, dtype: int64

In [ ]:
df["Has Online delivery"].value_counts()

Has Online delivery
No     7091
Yes    2451
Name: count, dtype: int64

In [ ]:
df.groupby("Has Table booking")["Aggregate rating"].mean()

Has Table booking
No     2.557956
Yes    3.441969
Name: Aggregate rating, dtype: float64

In [ ]:
df.groupby("Has Online delivery")["Aggregate rating"].mean()

Has Online delivery
No     2.463517
Yes    3.248837
Name: Aggregate rating, dtype: float64

In [ ]:
df.groupby("Price range")["Has Online delivery"].value_counts()

Price range  Has Online delivery
1            No                     3737
             Yes                     701
2            No                     1827
             Yes                    1286
3            No                      994
             Yes                     411
4            No                      533
             Yes                      53
Name: count, dtype: int64

# Level 2 Task 2 - Price Range Analysis

In [ ]:
avg_rating = df.groupby("Price range")["Aggregate rating"].mean()
print(avg_rating)

Price range
1    1.997476
2    2.941054
3    3.682633
4    3.817918
Name: Aggregate rating, dtype: float64


In [ ]:
print(avg_rating.idxmax())

4


In [ ]:
df[df["Price range"]==1]["Rating color"].value_counts()

Rating color
Orange        1895
White         1700
Yellow         606
Green          144
Red             62
Dark Green      31
Name: count, dtype: int64

In [ ]:
df[df["Price range"]==2]["Rating color"].value_counts()

Rating color
Orange        1425
Yellow         794
White          402
Green          325
Red             98
Dark Green      69
Name: count, dtype: int64

In [ ]:
df[df["Price range"]==3]["Rating color"].value_counts()

Rating color
Yellow        496
Green         415
Orange        313
Dark Green    126
White          35
Red            20
Name: count, dtype: int64

In [ ]:
df[df["Price range"]==4]["Rating color"].value_counts()

Rating color
Yellow        200
Green         194
Orange        101
Dark Green     74
White          11
Red             6
Name: count, dtype: int64

## Price Range Analysis
Restaurant ratings increase with price range. Lower-priced restaurants are predominantly associated with Orange ratings, while higher-priced restaurants are predominantly associated with Yellow ratings. Price Range 4 restaurants achieve the highest average rating (3.82), indicating a positive relationship between pricing and customer satisfaction.

# Level 2 Task 3 - Feature Engineering

In [ ]:
df['name_length'] = df['Restaurant Name'].apply(len)

In [ ]:
df['Has Table booking encoded'] = df['Has Table booking'].map({'Yes': 1, 'No': 0})

In [ ]:
df['Has Online delivery encoded'] = df['Has Online delivery'].map({'Yes': 1, 'No': 0}) 

In [ ]:
df.to_csv("../data/Processed_Dataset.csv", index=False)